src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestWhile.qs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Testing.QIR
+{
+    function TestWhile (a : Int, b : Int) : Int
+    {
+        mutable n = a;
+        while (n < b)
+        {
+            set n = n * 2;
+        }
+        return n;
```

Comment:
Just FYI: 
For some combinations of `a` and `b` the code has a risk of looping forever. E.g. if value of `b` is close to maximum possible (e.g. `Int` is 2 bytes in size and value of `b` is 0x7FFF then all the values of `n` may be overflowing from positive values below 0x7FFF down to the negative values).

Reply:
Interesting - I hadn't looked too closely at the example beyond whether it translates correctly. Given that none of the tests is actually executing and what is tested is merely the translation of different language constructs, I'll leave it as is.